More engine related renaming and re-organizing

Berthalamew opened the pull request:
Try and organize some of the Engine related code we have into the Blam/Engine folder
rename structs so that they try and parody original struct names
move some of the structs/definitions to proper modules

This is beneficial since it tries to keep everything game related contained to a single folder and tries to parody the original layout and naming.

Berthalamew committed:
move filo to files_windows module

Berthalamew committed:
remove Engine module and move functions to proper locations

- create game_engine_util module
- create life_cycle_manager module
- move e_game_life_cycle to life_cycle_manager header
- update logic that used to use the Engine module
fix compile

Berthalamew committed:
remove filo naming scheme

replace with more accurate naming scheme

Berthalamew committed:
move strings modules to Blam folder

- string_ids.h technically isn't it's own module in the original source but might as well keep it separated

Berthalamew committed:
string type renames

rename tag_string to static_string
rename tag_unicode to static_wchar_string

Berthalamew committed:
rename typedefs and function pointers for consistency

nukeulater commented on xlive/Blam/Engine/Simulation/MachineID.h:
```diff
@@ -2,5 +2,13 @@
 #pragma once
 struct s_machine_identifier
 {
-	char gap[6];
```
this is probably the abEnet, just do `byte machine_identifier[6]`

Himanshu-01 replied:
Yea its actually abNet only but bungie used `s_machine_identfier` as their goto structure in a lot of places instead declaring a 6 byte long array everywhere so we should keep this structure in my opinion

nukeulater replied:
meant it should actually be defined as `byte machine_identifier[6]` in that structure

Berthalamew replied:
done

nukeulater commented:
other than those unnecessary reverted commits present in this PR: https://github.com/pnill/cartographer/pull/426/commits/84b4b98b977cdb8d82a27247e2e1f46d009245f5 https://github.com/pnill/cartographer/pull/426/commits/3a454c901a03e0221808a19830a7888012799011

lgtm for the most part, and if @Himanshu-01 doesn't have anything to add.

Himanshu-01 commented on xlive/Blam/Engine/game/local_players.h:
```diff
@@ -1,7 +1,38 @@
+CHECK_STRUCT_SIZE(s_player_identifier, 8);
+
+#pragma pack(push, 1)
+struct s_local_player_properties
```
Actual name for this is `s_session_interface_globals` which comes under `networking` not `game `

Berthalamew replied:
renamed, and moved updated definition to NetworkSession.h

Himanshu-01 commented on xlive/H2MOD/Modules/Tweaks/Tweaks.cpp:
```diff
@@ -2,7 +2,7 @@
 
 #include "Tweaks.h"
 #include "Blam/Engine/game/game_time.h"
-#include "Blam\FileSystem\FiloInterface.h"
```
Would recommend maintaining a consistency during file inclusion 
Either use forward slashes `\` or or backward slashes  `/` for example this file has all the combinations messed up

```C
#include "Blam/Engine/tag_files/files_windows.h"
#include "Blam\Engine\Networking\NetworkMessageTypeCollection.h"
```

Himanshu-01 replied:
While you are at this . I would suggest getting rid of this namespace as well inside `objects/objects.h`

```c++
namespace Engine::Objects
{
	void create_new_placement_data(s_object_placement_data* object_placement_data, datum object_definition_idx, datum object_owner_idx, int a4);
	datum object_new(s_object_placement_data* object_placement_data);
	void apply_biped_object_definition_patches();
	void simulation_action_object_create(datum object_idx);
	void object_destroy(datum object_idx);
	int object_get_count();
	int object_count_from_iter();
}
```

Berthalamew replied:
Will remove that namespace in another PR,

Berthalamew replied:
> 
Changed includes to use forward slashes in tweaks

Berthalamew committed:
add requested changes

- add s_session_interface_globals and s_session_interface_user definitions and remove similar ones from local_players
- cleanup header references in Tweaks to use forward slashes

Berthalamew committed:
modify s_machine_identifier

change datatype
